armi/reactor/parameters/parameterCollections.py

```diff
@@ -493,6 +494,32 @@ def restoreBackup(self, paramsToApply):
                 pd.assigned = SINCE_ANYTHING
                 self.assigned = SINCE_ANYTHING
 
+    def where(
+        self, f: Callable[[parameterDefinitions.Parameter], bool]
+    ) -> Iterator[parameterDefinitions.Parameter]:
+        """Produce an iterator over parameters that meet some criteria.
+
+        Parameters
+        ----------
+        f : callable function f(parameter) -> bool
+            Function to check if a parameter should be fetched during the iteration.
+
+        Returns
+        -------
+        iterator of :class:`armi.reactor.parameters.Parameter`
+            Iterator, **not** list or tuple, that produces each parameter that
+            meets ``f(parameter) == True``.
+
+        Examples
+        --------
+        >>> block = r.core[0][0]
+        >>> pdef = block.p.paramDefs
+        >>> for param in pdef.where(lambda pd: pd.atLocation(ParamLocation.EDGES)):
+        ...     print(param.name, block.p[param.name])
+
```

Comment:
UNIMPORTANT

I see you put these empty lines at the end of a about half your docstrings. I just checked, the RST renders fine without this.

Reply:
Old habits not consistency implemented I guess 😅How would you like to proceed?

<details>

<summary> Rabbit hole research on conflicting and inconsistent docstring standards </summary>

The trailing empty line in a `numpydoc` is an implied but not explicit part of the standard. All of the examples provided in [the standard](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) have the trailing line included. But it's not stated as a part of the spec and things work well with / without it.

But also, the standard includes this line [towards the end](https://numpydoc.readthedocs.io/en/latest/format.html#common-rest-concepts):

> Line spacing and indentation are significant and should be carefully followed

The [Python docstring conventions PEP](https://peps.python.org/pep-0257/#multi-line-docstrings) also includes

> Insert a blank line after all docstrings (one-line or multi-line) that document a class

</details>

Reply:
Eww... Well, without that knowledge, we have manually scrubbed all such blank lines from ARMI, because the RST renders fine without them and empty lines don't seem helpful.

I can write a script to add them all back in, if this becomes necessary.

Thanks for the info!